OBS fails to build with CEF5060 due to #475 + #478 , and #477

### Operating System Info

Other

### Other OS

Devuan Linux (Debian variant)

### OBS Studio Version

Git

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

N/A

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

Successful build using CEF5060

### Current Behavior

Building with CEF 6533 completes without issue, building with CEF 5060 fails as follows;

```
[ 56%] Building CXX object plugins/obs-browser/CMakeFiles/obs-browser.dir/browser-client.cpp.o
In file included from /home/builder/obs/obs-port-31.0.0-153-g95dd4e9b3_5060_0/obs-studio/plugins/obs-browser/browser-client.cpp:19:
/home/builder/obs/obs-port-31.0.0-153-g95dd4e9b3_5060_0/obs-studio/plugins/obs-browser/browser-client.hpp:101:22: error: ‘virtual void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, CefRequestHandler::TerminationStatus, int, const CefString&)’ marked ‘override’, but does not override
  101 |         virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status, int error_code,
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~

```
Reverting #475 + #478 (commits 52c1527 and 46adc4b) solves this, but then #477 causes another failure;
```
/home/builder/obs/obs-port-31.0.0-153-ga269479c8_5060_0/obs-studio/plugins/obs-browser/obs-browser-plugin.cpp: In function ‘void BrowserInit()’:
/home/builder/obs/obs-port-31.0.0-153-ga269479c8_5060_0/obs-studio/plugins/obs-browser/obs-browser-plugin.cpp:386:91: error: ‘CefGetExitCode’ was not declared in this scope
  386 |                 blog(LOG_ERROR, "[obs-browser]: CEF failed to initialize. Exit code: %d", CefGetExitCode());
      |                                                                                           ^~~~~~~~~~~~~~

```

After additionally reverting #477 (commit af0651b) the build completes successfully with CEF5060, and runs as expected.


### Steps to Reproduce

1. Build obs-studio from current git master using CEF 5060
2.
3.
...


### Anything else we should know?

I had the impression that building with CEF 5060 is still intended to be supported as a stable fallback? Given issues reported with newer CEFs, and was building both to compare for my use case.
 
Build options used;
-DENABLE_RELOCATABLE=ON 
-DENABLE_PORTABLE_CONFIG=ON
-DCMAKE_BUILD_TYPE="Release" 
-DOBS_COMPILE_DEPRECATION_AS_WARNING=ON
-DCMAKE_INSTALL_PREFIX="${inst_parent}/obs-${obs_ver}" 
-DCEF_ROOT_DIR="${src_repo}/CEF_lib/${CEF_file}"
-DLibDataChannel_DIR="${build_parent}/3plib/ldc/lib/cmake/LibDataChannel"
-DENABLE_BROWSER=ON 
-DENABLE_BROWSER_PANELS=ON
-DENABLE_WEBRTC=ON
-DENABLE_RTMPS=ON 
-DENABLE_NEW_MPEGTS_OUTPUT=ON
-DENABLE_SERVICE_UPDATES=ON
-DENABLE_PULSEAUDIO=ON 
-DENABLE_PIPEWIRE=ON 
-DENABLE_LIBFDK=ON
-DENABLE_DECKLINK=OFF 
-DENABLE_VLC=OFF 
-DENABLE_JACK=OFF 
-DENABLE_SPEEXDSP=OFF
-DENABLE_RNNOISE=OFF 
-DENABLE_VST=OFF 
-DENABLE_OSS=OFF
-DENABLE_AJA=OFF

--- a/browser-client.cpp
+++ b/browser-client.cpp
@@ -83,13 +83,21 @@ CefRefPtr<CefResourceRequestHandler> BrowserClient::GetResourceRequestHandler(Ce
 	return nullptr;
 }
 
-void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, TerminationStatus, int,
-					      const CefString &error_string)
+void BrowserClient::OnRenderProcessTerminated(CefRefPtr<CefBrowser>, TerminationStatus
+#if CHROME_VERSION_BUILD >= 6367
+					      ,
+					      int, const CefString &error_string
+#endif
+)
 {
 	if (!valid())
 		return;
 
+#if CHROME_VERSION_BUILD >= 6367
 	std::string str_text = error_string;
+#else
+	std::string str_text = "<unknown>";
+#endif
 
 	const char *sourceName = "<unknown>";
 

--- a/browser-client.hpp
+++ b/browser-client.hpp
@@ -98,8 +98,12 @@ public:
 	GetResourceRequestHandler(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame,
 				  CefRefPtr<CefRequest> request, bool is_navigation, bool is_download,
 				  const CefString &request_initiator, bool &disable_default_handling) override;
-	virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status, int error_code,
-					       const CefString &error_string) override;
+	virtual void OnRenderProcessTerminated(CefRefPtr<CefBrowser> browser, TerminationStatus status
+#if CHROME_VERSION_BUILD >= 6367
+					       ,
+					       int error_code, const CefString &error_string
+#endif
+					       ) override;
 
 	/* CefResourceRequestHandler */
 	virtual CefResourceRequestHandler::ReturnValue OnBeforeResourceLoad(CefRefPtr<CefBrowser> browser,

--- a/obs-browser-plugin.cpp
+++ b/obs-browser-plugin.cpp
@@ -383,7 +383,11 @@ static void BrowserInit(void)
 #endif
 
 	if (!success) {
+#if CHROME_VERSION_BUILD >= 6367
 		blog(LOG_ERROR, "[obs-browser]: CEF failed to initialize. Exit code: %d", CefGetExitCode());
+#else
+		blog(LOG_ERROR, "[obs-browser]: CEF failed to initialize.");
+#endif
 		return;
 	}
 

